source/client-side-operations-timeout/tests/command-execution.yml

```diff
@@ -39,7 +39,7 @@ tests:
               failCommands: ["hello", "isMaster"]
```

Comment:
The 10 sample minimum is an attempt to reduce the probability that early RTT outliers cause unnecessary operation skipping. Keep in mind that the "min RTT" logic currently in the Go driver was not added as part of CSOT, but was an optimization added with Go driver connection pool improvements. I decided to be conservative with the implementation because it was a novel and untested concept at the time and impacts all Go driver users.

We haven't encountered any known issues with that "min RTT" implementation, so it makes sense to be less conservative with the opt-in CSOT behavior. However, trusting the first RTT sample on startup seems dangerous and may lead to to increased operation errors immediately after connecting, until more RTT samples are collected. I recommend we collect at least 2 samples before assuming it's an accurate representation of the actual RTT.

I don't have any stats on the impact of requiring 1 vs 2 RTT samples, only the observation that [networks are unreliable](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing#The_fallacies) and that any single RTT sample may have a nontrivial probability of being bogus. That may be especially true while an application is starting up.

As far as using a fixed number of samples (e.g. 10), that seems totally reasonable.

Reply:
Updated the spec to use a min of 2 and max of 10 samples. I also updated the tests to wait (heuristically) for >=2 samples.